### PR TITLE
silence setHyperPars

### PR DESCRIPTION
**Pull request by mllg**

### CODE REVIEW

**Comment by berndbischl** [comment]
@giuseppec 
michel might be busy. pls take over.

but this must be done:
1) travis must run

2) when on.par.without.desc = quiet we dont want to see the hyperpar suggestion message from the string matching

**Comment by berndbischl** [comment]
this still procduces the message

configureMlr(on.par.without.desc = "quiet")
lrn = makeLearner("classif.ksvm", sigm=1, show.info = T)

also make sure to unit test properly. i would guess you also want to use a mocklearner here to make sure the passed param actually reaches the learner.

and you need to test that the message is NOT generated, too. i hope test that can do that.
(i think so)

**Comment by giuseppec** [comment]
I think adding `show.info` arg in `setHyperPars` is not good because you can already set `show.info` on learner creation and something like this would be possible:
```
wl = makeLearner("classif.ksvm", config = list(on.par.without.desc = "warn", show.info = TRUE), 
  sigm = 1, show.info = FALSE)
```
@mllg why do you want the show.info in setHyperPars? Isn't it sufficient to set the `show.info` using the `config` when creating the learner?

File: R/makeLearner.R [hunk 1]
```diff
@@ -55,16 +55,19 @@ makeLearner = function(cl, id = cl, predict.type = "response", predict.threshold
 
   assertString(cl)
   assertFlag(fix.factors.prediction)
+  assertList(config, names = "named")
+  # FIXME: maybe forbit show.info here issue #1098:
+  assertSubset(names(config), choices = names(formals(configureMlr)))
```
Comment (mllg) [inline R/makeLearner.R]: names(getMlrOptions())?

File: tests/testthat/test_base_hyperpars.R [hunk 2]
```diff
@@ -98,5 +133,8 @@ test_that("options are respected", {
   configureMlr(on.par.out.of.bounds = "quiet")
   expect_is(setHyperPars(lrn, alpha = 2), "Learner")
 
+  configureMlr(on.par.without.desc = "quiet")
+  lrn = makeLearner("classif.ksvm", sigm = 1, show.info = TRUE)
```
Comment (mllg) [inline tests/testthat/test_base_hyperpars.R]: expect_silent()?

File: tests/testthat/test_base_hyperpars.R [hunk 3]
```diff
@@ -63,12 +63,51 @@ test_that("setting 'when' works for hyperpars", {
...
+  cs = list(on.par.without.desc = "stop")
+  # never print message when quiet, no matter hat show.info is
+  expect_silent(makeLearner("classif.ksvm", config = cq, sigm = 1, show.info = TRUE))
+  expect_silent(makeLearner("classif.ksvm", config = cq, sigm = 1, show.info = FALSE))
+  configureMlr(on.par.without.desc = "quiet")
+  expect_silent(makeLearner("classif.ksvm", sigm = 1, show.info = TRUE))
+  expect_silent(makeLearner("classif.ksvm", sigm = 1, show.info = FALSE))
+
+  # show.info on/off and warn
+  expect_message(expect_warning(makeLearner("classif.ksvm", config = cw, sigm = 1, show.info = TRUE),
```
Comment (larskotthoff) [inline tests/testthat/test_base_hyperpars.R]: Why are `expect_message`/`expectWarning` etc nested here? Is this necessary?

**Comment by larskotthoff** [comment]
Looks good to me apart from the nested expectations that I mentioned above.

**Comment by berndbischl** [comment]
should be merged when travis is done.

but we need NEWS then for the show.info change.

**Pull request merged**